plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -2026,6 +2035,7 @@ private void assertDynamicFiltering(@Language("SQL") String sql, JoinDistributio
                 noDynamicFilteringResultWithQueryId.getResult(),
                 "For query: \n " + sql);
 
+        // Flaky: Reuses global QueryTracker state which can pruneExpiredQueries()
```

Comment:
it's not flaky in traditional sense. The query history gets purged so once this assertion fails it will keep failing even on retries. the other option is to retry entire test (either via `@Flaky` or wrapping the entire test in assertEventually).

Reply:
let's wrap test in assertEventually

Reply:
assertEventually is widely used but it does not have maxRetries. 🤷 
Does it worth to use failsafe here?

Reply:
assertEventually has a timeout duration. it should be enough

Reply:
Initially i understood `timeout` in a wrong way. It's for all tries, not the single one. So it's acceptable.

Reply:
I don't like usage of `assertEventually` because you retry entire test and not only wait for certain event that may happen eventually after the "when" phase of the test. In such this case I think usage of `@Flaky` is more  justified.

I believe here using `executeExclusively` is the best as that eliminates impact of any other test.

Reply:
`assertEventually` and `@Flaky` have the same logical meaning. 
They retry test. 
Without test retry the correct result will not be obtained `eventually`.
So `assertEventually` and `@Flaky` are equally incorrect.
I agree that `executeExclusively` would be a better fit. So I'll try that.

Reply:
@kokosing `executeExclusively` would work only if all other tests which use the same `runner` would check for the lock too (e.g. obtain a read lock). This is not a case here. 
Also we are interested in `executeExclusively` tests which require `QueryTracker` not only between themselves, but exclusively to all other tests as well. 
```
    protected void executeExclusively(Runnable executionBlock)
    {
        runner.getExclusiveLock().lock();
        try {
            executionBlock.run();
        }
        finally {
            runner.getExclusiveLock().unlock();
        }
    }
```

So I'll stick with `assertEventually` implementation.

Also, as an alternative solution to choose I can propose to annotate methods with:
```
@Flaky
@Test(priority = LAST)
```
@kokosing @hashhar @wendigo please choose what do you like more.

Reply:
That's why I prefer this more: https://github.com/trinodb/trino/pull/18468